src/parse.c

```diff
@@ -10467,10 +10469,145 @@ static tree_t p_psl_directive(void)
    return t;
 }
 
-static tree_t p_psl_clock_declaration(void)
+static type_t p_psl_subtype_indication(void)
+{
+   BEGIN("PSL subtype indication");
+
+   // Handle ambiguity in "p_subtype_indication". Two consecutive IDs
+   // may indicate resolution function, or only "type_mark" followed by
+   // actual parameter name.
```

Comment:
Couldn't you just call `p_subtype_indication()` here? Because subtype indication is `[ name ] type_mark [ constraint ]`.

Reply:
I think no. When you put:
`const hdltype integer my_fancy_parameter`

then `p_subtype_indication` will evaluate:
`if ((peek() == tID && peek_nth(2) == tID) || peek() == tLPAREN) `
as true, and try to parse `integer` as subtype. Instead, it should
parse `p_type_mark`.

I wanted to avoid changing `p_subtype_indication`.

Reply:
OK I see. It's not a problem in VHDL because the name comes before the type.